libvast/include/vast/concept/parseable/vast/pattern.hpp

```diff
@@ -29,7 +29,12 @@ struct access::parser_base<pattern>
 
   template <class Iterator>
   bool parse(Iterator& f, const Iterator& l, pattern& a) const {
-    return pattern_parser{}(f, l, a.str_);
+    if (!pattern_parser{}(f, l, a.str_)) {
+      return false;
+    }
+    auto case_insens_parser = parsers::chr{pattern::case_insensitive_flag};
+    a.case_insensitive_ = case_insens_parser(f, l, unused);
+    return true;
```

Comment:
The way this is implemented, `pattern_parser{}(...)` and `to<pattern>(...)` behave differently, because the former does not support the optional `i` flag. Please unify this.

Reply:
Can you elaborate on where the divergence lies, because I tried to unit test `to<pattern>` and all checks pass:

```cpp
TEST(to pattern) {
  auto p1 = to<pattern>("/test/");
  CHECK(p1);
  CHECK_EQUAL(p1->string(), "test");
  CHECK(!p1->case_insensitive());
  auto p2 = to<pattern>("/test/i");
  CHECK(p2);
  CHECK_EQUAL(p2->string(), "test");
  CHECK(p2->case_insensitive());
}
```

Reply:
Here's an example that fails:

```cpp
auto str = std::string{"/foo/i"};
auto p = pattern_parser{};
auto result = pattern{};
CHECK(p(str, result)); // this fails
```

This is because the `pattern_parser` itself does not know about the flags, which is bad when you want to combine it. Now this isn't used anywhere in the code base currently, but it's still an easy trap to fall into. I see two possible fixes:
- Rename `pattern_parser` to `slash_delimited_string`, because that's what the parser is doing.
- Change the `pattern_parser` to also include the optional `/i` suffix and have it parse into a pattern.

I believe the former is more correct.

Reply:
Ahh, the problem is much simpler than I thought then. I renamed the `pattern` parsers in https://github.com/tenzir/vast/pull/2951/commits/0a05c366639d88bfa94d71648e95a2bdd95361c2 and added an integration test in https://github.com/tenzir/vast/pull/2951/commits/7d6238fd08206cfe587317772ebbe46811ef005c